### PR TITLE
Simplify SDL_BLENDMODE_MUL

### PR DESCRIPTION
**Pull request by 1bsyl**
see https://github.com/libsdl-org/SDL/issues/7473

### CODE REVIEW

File: src/video/sdlgenblit.pl [hunk 1]
```diff
@@ -398,7 +398,6 @@ sub output_copycore
 __EOF__
             } else {
```
Comment (slouken) [inline src/video/sdlgenblit.pl:399]: You can remove this entire else branch

File: src/video/SDL_blit_auto.c [hunk 2]
```diff
@@ -1103,34 +1019,20 @@ static void SDL_Blit_RGB888_ARGB8888_Modulate_Blend(SDL_BlitInfo *info)
...
-                dstB = ((srcB * dstB) + (dstB * (255 - srcA))) / 255; if (dstB > 255) {
-                    dstB = 255;
-                }
-                dstA = ((srcA * dstA) + (dstA * (255 - srcA))) / 255; if (dstA > 255) {
-                    dstA = 255;
-                }
+                dstR = ((srcR * dstR) + (dstR * (255 - srcA))) / 255; if (dstR > 255) dstR = 255;
+                dstG = ((srcG * dstG) + (dstG * (255 - srcA))) / 255; if (dstG > 255) dstG = 255;
+                dstB = ((srcB * dstB) + (dstB * (255 - srcA))) / 255; if (dstB > 255) dstB = 255;
+                dstA = ((srcA * dstA) + (dstA * (255 - srcA))) / 255; if (dstA > 255) dstA = 255;
```
Comment (slouken) [inline src/video/SDL_blit_auto.c]: Why do we still have a dstA operation here?

File: include/SDL3/SDL_blendmode.h [hunk 3]
```diff
@@ -52,7 +52,8 @@ typedef enum
                                               dstA = dstA */
     SDL_BLENDMODE_MUL = 0x00000008,      /**< color multiply
                                               dstRGB = (srcRGB * dstRGB) + (dstRGB * (1-srcA))
-                                              dstA = (srcA * dstA) + (dstA * (1-srcA)) */
+                                              dstA = (srcA * dstA) + (dstA * (1-srcA))
```
Comment (slouken) [inline include/SDL3/SDL_blendmode.h:55]: This should just say dstA = dstA, like the other operations

File: src/video/sdlgenblit.pl (hunk 1 above)
Comment (1bsyl) [reply src/video/sdlgenblit.pl]: removed

File: src/video/SDL_blit_auto.c (hunk 2 above)
Comment (1bsyl) [reply src/video/SDL_blit_auto.c]: I refreshed the SDL_blit_auto.c file before applying the real modification fro sdlgenblit.pl

File: include/SDL3/SDL_blendmode.h (hunk 3 above)
Comment (1bsyl) [reply include/SDL3/SDL_blendmode.h]: yes

File: src/render/software/SDL_draw.h [hunk 4]
```diff
@@ -91,6 +91,7 @@
 #define DRAW_SETPIXEL_MUL(getpixel, setpixel)      \
     do {                                           \
         unsigned sr, sg, sb, sa;                   \
+        (void)sa;                                  \
```
[the remark this answers is not in the document]
Comment (1bsyl) [reply src/render/software/SDL_draw.h]: bunch of warning without this :/
https://github.com/libsdl-org/SDL/actions/runs/4434977964/jobs/7781556284?pr=7475

```
D:/a/SDL/SDL/src/render/software/SDL_blendline.c: In function 'SDL_BlendLine_RGB2':
D:/a/SDL/SDL/src/render/software/SDL_draw.h:93:30: error: variable 'sa' set but not used [-Werror=unused-but-set-variable]
   93 |         unsigned sr, sg, sb, sa;                   \
      |                              ^~
D:/a/SDL/SDL/src/render/software/SDL_draw.h:380:13: note: in definition of macro 'HLINE'
  380 |             op;                                                \
      |             ^~
D:/a/SDL/SDL/src/render/software/SDL_draw.h:286:5: note: in expansion of macro 'DRAW_SETPIXEL_MUL'
  286 |     DRAW_SETPIXEL_MUL(RGB_FROM_PIXEL(*pixel, fmt, sr, sg, sb), \
      |     ^~~~~~~~~~~~~~~~~
D:/a/SDL/SDL/src/render/software/SDL_blendline.c:61:27: note: in expansion of macro 'DRAW_SETPIXEL_MUL_RGB'
   61 |             HLINE(Uint16, DRAW_SETPIXEL_MUL_RGB, draw_end);
      |                           ^~~~~~~~~~~~~~~~~~~~~
D:/a/SDL/SDL/src/render/software/SDL_draw.h:93:30: error: variable 'sa' set but not used [-Werror=unused-but-set-variable]
   93 |         unsigned sr, sg, sb, sa;                   \
      |                              ^~
```

[the remark this answers is not in the document]
Comment (1bsyl) [reply src/render/software/SDL_draw.h:94]: Also, I removed the 
sa = 0xff
this match BLENDMODE_MOD. with also sets dstA = dstA

**Comment by slouken** [comment]
Can you squash this for review?

**Comment by 1bsyl** [comment]
I think it's better not to squash all because. there are 1 commit just to refresh the blit_auto before, and one after also. but I can if really needed
( to see the full diff at once, there is the tab "Files changed" ! )

**Comment by 1bsyl** [comment]
ok let's merge !

**Pull request merged**